modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java

```diff
@@ -310,11 +310,20 @@ public void destroy(){
 
 	@Override
 	public void destroy(boolean preserveState) {
+		destroy(preserveState, false);
+	}
+
+	@Override
+	public void destroy(boolean preserveState , boolean isArtifactUpdate) {
 
 		if (!preserveState) {
-			stop();
 			deleteMessageProcessorState();
 		}
+		if (isArtifactUpdate || !preserveState) {
+			/*	All the tasks need to be stopped during artifact update and re deployed so that the changes in member
```

Comment:
There seems to be an indentation issue

Reply:
This is because the previous one is intended with tabs, I tried formatting but it generates a big diff. However this shows up correctly in idea.